scripts/utils/docs-content-tools/handlers.js

```diff
@@ -0,0 +1,142 @@
+const glob = require('glob');
+const frontmatterGH = require('@github-docs/frontmatter');
+const fs = require('fs');
+
+/**
+ * Adds slashes to beginning and end of directory if missing
+ * @param {String} directory - The directory to add slashes to
+ * @returns {String} The directory with slashes added if necessary
+ */
+const formatDirectory = (directory) => {
+  !directory.startsWith('/') && (directory = `/${directory}`);
+  !directory.endsWith('/') && (directory = directory.concat('/'));
+  return directory;
+};
+
+/**
+ * Gets slug from path, relative to contents folder
+ * @param {String} filepath - The full filepath of the file
+ * @returns {String} The slug of the filepath
+ */
+const getSlugFromPath = (filepath) => {
+  return filepath.match(/content(.*?).mdx/)[1];
+};
+
+/**
+ * Reads a file and separates the content and frontmatter data
+ * @param {String} path - The full filepath of the file
+ * @returns {Object} Original path, slug from path, content, frontmatter data
+ */
+const separateData = async (path) => {
+  const file = fs.readFileSync(path, 'utf8');
+  const slug = getSlugFromPath(path);
+  const separateData = await frontmatterGH(file);
+  const { data: frontmatterData, content } = separateData;
+  if (!frontmatterData.redirects) {
+    frontmatterData.redirects = [];
+  }
+  return { path, content, frontmatterData, slug };
+};
+
+/**
+ * Joins the content and frontmatter data
+ * @param {Object} data - content, frontmatterData, path
+ * @returns {Object} Combined data and path
+ */
+const joinData = ({ content, frontmatterData, path }) => {
+  const data = frontmatterGH.stringify(content, frontmatterData, {
+    lineWidth: -1,
+  });
+
+  return { path, data };
+};
+
+/**
+ * Adds a redirect if the slug is not already in the redirects array
+ * @param {Object} data - slug, frontmatterData, rest
+ * @returns {Object} Edited frontmatterData, whether the redirect was added or skipped
+ */
+const addRedirect = ({ frontmatterData, slug, ...rest }) => {
+  if (frontmatterData.redirects.includes(slug)) {
+    return { frontmatterData, ...rest, skipped: slug };
+  }
+  frontmatterData.redirects.push(slug);
```

Comment:
I'm pretty sure that [`Array.prototype.push`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) modifies the original array, which might cause some weird side-effects. Here's what I mean by that:

```js
const me = { name: 'Zack', favNums: [1, 2, 3] };

const func = (obj) => {
  obj.favNums.push(4);
  return obj;
};

console.log(me); // { name: 'Zack', favNums: [1, 2, 3] };

// the result contains the `push`ed infromation
func(me); // returns { name: 'Zack', favNums: [1, 2, 3, 4] };

// the original array _also_ contains the `push`ed information
console.log(me); // { name: 'Zack', favNums: [1, 2, 3, 4] };
```

I'm not sure if that will cause a problem, but it has the potential to be a nasty bug that could be hard to track down. I'd suggest using a [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) (since it creates a new array) when you return:

```js
return {
  frontmatterData: {
    ...frontmatterData,
    redirects: [...frontmatterData.redirects, slug],
  },
  ...rest,
  added: slug,
};
```